### PR TITLE
align fuzzing deps

### PR DESCRIPTION
**Pull request by ilanpillemer**
Signed-off-by: Ilan <ilan.pillemer@gmail.com>

@pjbgf as discussed in https://github.com/fluxcd/flux2/issues/2552

### CODE REVIEW

File: tests/fuzz/oss_fuzz_build.sh [hunk 1]
```diff
@@ -89,7 +89,7 @@ else
     curl -s --fail https://raw.githubusercontent.com/fluxcd/image-reflector-controller/${REFLECTOR_VER}/config/crd/bases/image.toolkit.fluxcd.io_imagepolicies.yaml -o testdata/crds/imagepolicies.yaml
 fi
 
-go mod tidy -compat=1.17
+go get -d github.com/AdaLogics/go-fuzz-headers -compat=1.17
```
Comment (pjbgf) [inline tests/fuzz/oss_fuzz_build.sh:92]:
```suggestion
go get -d github.com/AdaLogics/go-fuzz-headers
```

File: tests/fuzz/oss_fuzz_build.sh [hunk 2]
```diff
@@ -54,7 +54,7 @@ export PKG_CONFIG_PATH="${TARGET_DIR}/lib/pkgconfig:${TARGET_DIR}/lib64/pkgconfi
 export CGO_CFLAGS="-I${TARGET_DIR}/include -I${TARGET_DIR}/include/openssl"
 export CGO_LDFLAGS="$(pkg-config --libs --static --cflags libssh2 openssl libgit2)"
 
-go mod tidy -compat=1.17
+go get -d github.com/AdaLogics/go-fuzz-headers -compat=1.17
```
Comment (pjbgf) [inline tests/fuzz/oss_fuzz_build.sh:57]:
```suggestion
go get -d github.com/AdaLogics/go-fuzz-headers
```

**Comment by pjbgf** [comment]
@ilanpillemer you will need to sign-off the commit to pass the DCO check. Apart from that, it LGTM.

**Comment by ilanpillemer** [comment]
@pjbgf Does the below `go get` need to also be added to same file?

`image_update_fuzzer.go:35:2: module github.com/fluxcd/image-automation-controller provides package github.com/fluxcd/image-automation-controller/controllers and is replaced but not required; to add it:
	go get github.com/fluxcd/image-automation-controller
image_update_fuzzer.go:36:2: module github.com/fluxcd/image-automation-controller provides package github.com/fluxcd/image-automation-controller/pkg/update and is replaced but not required; to add it:
	go get github.com/fluxcd/image-automation-controller`

**Comment by pjbgf** [comment]
@ilanpillemer I think so, that's because of the amendments on the `go.mod` and the fact that we need to refer to the current version of the `image-automation-controller`.

Can you please also squash all the commits name it based on this PR's title?

**Pull request merged**